src/main/java/org/openlmis/requisition/service/RequisitionSecurityService.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+package org.openlmis.requisition.service;
+
+import org.openlmis.requisition.domain.Requisition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class RequisitionSecurityService {
+
+  @Autowired
+  private PermissionService permissionService;
+
+  /**
+   * Filters requisitions based on user permissions. It strives to make as little calls to the
+   * reference data service as possible.
+   *
+   * @param allRequisitions input list containing any requisitions
+   * @return filtered input list of requisitions, containing only those that the user has access to
+   */
+  public List<Requisition> filterInaccessibleRequisitions(List<Requisition> allRequisitions) {
+
+    Map<RightsFor, Boolean> verified = new HashMap<>();
```

Comment:
Extacly, if we were to save it for multiple requests we would have to consider cache invalidation. This prevents duplicate calls within a single search request (and we made a lot of them while only one right check is required for each facility and program)